samples/ODataAuthorizationSample/Startup.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading.Tasks;
+using AspNetCore3ODataPermissionsSample.Models;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Authorization;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AspNetCore3ODataPermissionsSample
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDbContext<AppDbContext>(opt => opt.UseLazyLoadingProxies().UseInMemoryDatabase("CustomerOrderList"));
+            services.AddOData();
+
+            // add OData authorization services
+            services.AddODataAuthorization((options) =>
+            {
+                // we setup a custom scope finder that will extract the user's scopes from the Permission claims
+                options.ScopesFinder = (context) =>
+                {
+                    var permissions = context.User?.FindAll("Permission");
+                    if (permissions == null)
+                    {
+                        return Task.FromResult(Enumerable.Empty<string>());
+                    }
+
+                    return Task.FromResult(permissions.Select(p => p.Value));
+                };
+
+                options.ConfigureAuthentication("AuthScheme")
+                    .AddScheme<CustomAuthentationOptions, CustomAuthenticationHandler>("AuthScheme", options => { });
+            });
+
+            // OData authorization depends on the AspNetCore authentication and authorization services
+            // we need to specify at least one authentication scheme and handler. Here we opt for a simple custom handler defined
+            // later in this file, for demonstration purposes. Could also use cookie-based or JWT authentication
```

Comment:
Might it be worth adding a line or 2 to show how someone might use cookie-based and/or JWT auth with this Auth lib?

Reply:
I added a separate demo that uses cookie-based auth, but it's on a different branch that uses the new model builder.

Reply:
Since the modelbuilder has been released, I can update that branch and merged it to the main branch.